src/Integration.Vsix/CFamily/CFamilyPreCompiledHeadersEventListener.cs

```diff
@@ -0,0 +1,103 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.ComponentModel.Composition;
+using System.IO;
+using System.Linq;
+using SonarLint.VisualStudio.Core.CFamily;
+using SonarLint.VisualStudio.Integration.Vsix.Helpers.DocumentEvents;
+
+namespace SonarLint.VisualStudio.Integration.Vsix.CFamily
+{
+    interface ICFamilyPreCompiledHeadersEventListener : IDisposable
+    {
+        void Listen();
+    }
+
+    [Export(typeof(ICFamilyPreCompiledHeadersEventListener))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    internal sealed class CFamilyPreCompiledHeadersEventListener : ICFamilyPreCompiledHeadersEventListener
+    {
+        internal const string PchJobId = "pch-generation";
+        internal const int PchJobTimeoutInMilliseconds = 60 * 1000;
+        internal const string PchFilePathSuffix = "SonarLintPCH.preamble";
+        internal string pchFilePath = Path.Combine(Path.GetTempPath(), PchFilePathSuffix);
+
+        private readonly ICLangAnalyzer cLangAnalyzer;
+        private readonly IActiveDocumentTracker activeDocumentTracker;
+        private readonly IScheduler scheduler;
+        private readonly ISonarLanguageRecognizer sonarLanguageRecognizer;
+        private bool disposed;
+
+        [ImportingConstructor]
+        public CFamilyPreCompiledHeadersEventListener(ICLangAnalyzer cLangAnalyzer,
+            IActiveDocumentTracker activeDocumentTracker, 
+            IScheduler scheduler,
+            ISonarLanguageRecognizer sonarLanguageRecognizer)
+        {
+            this.cLangAnalyzer = cLangAnalyzer;
+            this.activeDocumentTracker = activeDocumentTracker;
+            this.scheduler = scheduler;
+            this.sonarLanguageRecognizer = sonarLanguageRecognizer;
+        }
+
+        public void Listen()
```

Comment:
Is there an advantage in having 'Listen' as a separate method rather than just starting listening in the constructor?

Reply:
I think it's just odd to have an interface without any functions in it.

Reply:
Yep, but maybe that says something about the design or usage. What happens if `Listen` is called multiple times? Or `Listen` is called after `Dispose`?
It's not that uncommon to have a marker interface that doesn't do anything.